src/main/scala/com/databricks/labs/mosaic/functions/MosaicRegistry.scala

```diff
@@ -0,0 +1,59 @@
+package com.databricks.labs.mosaic.functions
+
+import com.databricks.labs.mosaic.expressions.base.WithExpressionInfo
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry
+import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionDescription, ExpressionInfo}
+import org.apache.spark.sql.catalyst.FunctionIdentifier
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry.FunctionBuilder
+
+import scala.collection.immutable
+import scala.reflect.runtime.universe
+import scala.reflect.ClassTag
+import scala.util.Try
+
+case class MosaicRegistry(registry: FunctionRegistry, database: Option[String] = None) {
```

Comment:
This class should simplify a lot the MosaicContext, but it is not used right now.
Is this intended?

Reply:
Yes, I didnt want to migrate vector APIs via raster PR.
It would touch all expression files.
We also have the vector expression simplification PR coming after raster is resolved.
We will use MosaicRegistry in the following PR.